android-components/components/browser/storage-sync/src/test/java/mozilla/components/browser/storage/sync/PlacesHistoryStorageTest.kt

```diff
@@ -545,15 +543,14 @@ class PlacesHistoryStorageTest {
         assertEquals(0, visits.size)
     }
 
-    @Ignore("Disabled: https://bugzilla.mozilla.org/show_bug.cgi?id=1853687")
     @Test
     fun `store can delete by 'range'`() = runTestOnMain {
         history.recordVisit("http://www.mozilla.org/1", PageVisit(VisitType.TYPED))
-        advanceUntilIdle()
+        Thread.sleep(10)
         history.recordVisit("http://www.mozilla.org/2", PageVisit(VisitType.DOWNLOAD))
-        advanceUntilIdle()
+        Thread.sleep(10)
         history.recordVisit("http://www.mozilla.org/3", PageVisit(VisitType.BOOKMARK))
-
+        Thread.sleep(10)
```

Comment:
I think we'll definitely want to avoid adding these sleeps to the tests. Though short, the less we impact our CI pipeline's run time the better. I submitted a [patch](https://github.com/mozilla-mobile/firefox-android/pull/4005) that I think will get these tests back on track, could you rebase on it once it lands? That way, we can revert my patch if it doesn't work and go back to ignoring these until we found a solution

Reply:
Thanks, yeah I figured it was probably bad but was curious if it sparked any ideas on simply ensuring we can get some sequential things going, since I _think_ it's now happening too fast. I don't mind holding off until that patch and will rebase after it lands.

Reply:
This has been rebased on your patch (btw, thanks for fixing that!)